### ISSUE TITLE
Possible buffer overflow?

### ISSUE THREAD

**Issue by EmielBruijntjes** [libpwquality/libpwquality#17]
When running in valgrind, the following error is reported:

```
==16811== Use of uninitialised value of size 8
==16811==    at 0x156C6976: read_config_file (settings.c:144)
==16811==    by 0x156C6D06: pwquality_read_config (settings.c:267)
```

When I look at the code in settings.c, I do indeed see some possible buffer overflows, for example:

```
len = strlen(linebuf);
if (linebuf[len - 1] != '\n' && !feof(f)) {
```

For the last empty line in the file len = 0, and linebuf[-1] is accessed.

```
do {
        if (!isspace(*(ptr-1))) {
                *ptr = '\0';
                break;
        }
        --ptr;
} while (ptr > linebuf);
```

ptr-1 may also refer to a memory address outside of the buffer, if I read this correctly.

**Comment by t8m** [comment]
The first one cannot happen - fgets returns NULL if no characters are read at the end of file.
On the other hand the second statement can really read one byte before the buffer - it is harmless error as it does not really affect anything. I'll fix it, thanks for the report.

**Comment by t8m** [comment]
Fixed in 9d6140b
